tokio/src/task/join_set.rs

```diff
@@ -295,7 +295,7 @@ impl<T: 'static> JoinSet<T> {
     ///
     /// This method is cancel safe. If `join_next_with_id` is used as the event in a `tokio::select!`
     /// statement and some other branch completes first, it is guaranteed that no tasks were
-    /// removed from this `JoinSet`.
+    /// removed from this `JoinSet`, nor any task's ID freed for re-use by another task.
```

Comment:
There are many methods needs this explanation, I prefer to add a dedicated section in the docs of `JoinSet`.